build_script.sh

```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 rm -f *.out 
-g++ -g -Wall -Wextra -pedantic main.cpp src/parser.cpp -o json_parser.out 
\ No newline at end of file
+g++ -g -Wall -Wextra -pedantic main.cpp src/parser.cpp src/json_obj.cpp src/json_value.cpp -o json_parser.out 
```

Comment:
This is why you need CMake. 
An improvement can be using src/*.cpp as a variable on top. For g++ & compiler flags, you can also do `CC=g++` & `FLAG=-g -Wall -Wextra -pedantic`, so you simply do `$CC $FLAGS main.cpp $SRC -o json_parser.out`, reusable and easy to update/maintain.

Reply:
The next thing I would do is moving this project to Cmake.